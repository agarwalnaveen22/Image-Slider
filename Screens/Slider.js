import React, {Component} from 'react';
import {
  RefreshControl,
  SafeAreaView,
  ScrollView,
  StyleSheet,
  ActivityIndicator,
} from 'react-native';
import {connect} from 'react-redux';
import {loadImagesRequest} from '../Actions';
import CommonMethods from '../Utilities/CommonMethods';
import ImageSlider from '../Components/ImageSliderComponent';

class Slider extends Component {
  constructor(props) {
    super(props);
    this.state = {
      images: [],
      refreshing: false,
    };
  }

  setData = async () => {
    this.props.loadImagesRequest({});
  };
  onRefresh = async () => {
    await this.setState({refreshing: true});
    let randomImages = CommonMethods.getRandomImages(this.props.appData.images);
    await this.setState({images: randomImages, refreshing: false});
  };
  async componentDidUpdate(prevProps) {
    if (prevProps.appData.images !== this.props.appData.images) {
      let randomImages = CommonMethods.getRandomImages(
        this.props.appData.images,
      );
      await this.setState({images: randomImages});
    }
  }
  componentDidMount() {
    this.setData();
  }
  render() {
    return (
      <SafeAreaView style={styles.container}>
        <ScrollView
          contentContainerStyle={styles.body}
          refreshControl={
            <RefreshControl
              refreshing={this.state.refreshing}
              onRefresh={() => this.onRefresh()}
            />
          }>
          {this.props.appData.loading ? (
            <ActivityIndicator
              size="large"
              color="#000000"
              style={styles.refreshBtn}
            />
          ) : (
            <ImageSlider
              images={this.state.images}
              sliderContainerStyle={styles.sliderContainer}
              sliderStyle={styles.slider}
              sliderImagesStyle={styles.sliderImages}
              bulletsStyle={styles.bullets}
              authorContainerStyle={styles.authorContainer}
              imageUrlId="imageModifedUrl"
            />
          )}
        </ScrollView>
      </SafeAreaView>
    );
  }
}

const styles = StyleSheet.create({
  container: {
    flex: 1,
    alignItems: 'center',
    marginTop: '40%',
    backgroundColor: '#ffffff',
  },
  body: {
    width: 220,
    height: 370,
    borderWidth: 1,
    borderRadius: 5,
  },
  refreshBtn: {
    top: '40%',
  },
  sliderContainer: {
    width: '100%',
    height: 340,
  },
  slider: {
    width: 220,
    height: 340,
  },
  sliderImages: {
    width: 200,
    height: 300,
    marginStart: 10,
    marginEnd: 10,
  },
  authorContainer: {
    width: '100%',
    height: 40,
    justifyContent: 'center',
    alignItems: 'center',
  },
  bullets: {
    fontSize: 15,
    color: '#000000',
  },
});

const mapStateToProps = (state) => {
  return state;
};

export default connect(mapStateToProps, {
  loadImagesRequest,
})(Slider);
