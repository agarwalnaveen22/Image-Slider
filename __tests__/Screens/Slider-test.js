import 'react-native';
import React from 'react';
import renderer from 'react-test-renderer';
import {Provider} from 'react-redux';
import configureMockStore from 'redux-mock-store';
import Slider from '../../Screens/Slider';
import imageData from '../ImageData.json';

const mockStore = configureMockStore();
const store = mockStore({});

describe('test slider page design and functionality', () => {
  var snap;
  const appData = {
    loading: false,
    images: imageData,
  };
  beforeEach(async () => {
    snap = await renderer.create(
      <Provider store={store}>
        <Slider appData={appData} />
      </Provider>,
    );
  });
  it('match snapshot', () => {
    expect(snap.toJSON()).toMatchSnapshot();
  });

  it('check methods onRefresh', async () => {
    await snap.toTree().rendered.rendered.instance.onRefresh();
    let images = snap.toTree().rendered.rendered.instance.state.images;
    expect(images.length).toEqual(5);
  });

  it('check methods componentDidUpdate', async () => {
    await snap
      .toTree()
      .rendered.rendered.instance.componentDidUpdate({appData: {images: []}});
    let images = snap.toTree().rendered.rendered.instance.state.images;
    expect(images.length).toEqual(5);
  });
});
