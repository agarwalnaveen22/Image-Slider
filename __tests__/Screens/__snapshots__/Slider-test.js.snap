// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test slider page design and functionality match snapshot 1`] = `
<RCTSafeAreaView
  emulateUnlessSupported={true}
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#ffffff",
      "flex": 1,
      "marginTop": "40%",
    }
  }
>
  <RCTScrollView
    contentContainerStyle={
      Object {
        "borderRadius": 5,
        "borderWidth": 1,
        "height": 370,
        "width": 220,
      }
    }
    refreshControl={
      <RefreshControlMock
        onRefresh={[Function]}
        refreshing={false}
      />
    }
  >
    <RCTRefreshControl />
    <View>
      <View>
        <View
          style={
            Array [
              Object {
                "alignItems": "center",
                "justifyContent": "center",
              },
              Object {
                "height": 340,
                "width": "100%",
              },
            ]
          }
        >
          <RCTScrollView
            data={Array []}
            disableVirtualization={false}
            getItem={[Function]}
            getItemCount={[Function]}
            horizontal={true}
            initialNumToRender={10}
            keyExtractor={[Function]}
            maxToRenderPerBatch={10}
            onContentSizeChange={[Function]}
            onEndReachedThreshold={2}
            onLayout={[Function]}
            onMomentumScrollEnd={[Function]}
            onScroll={[Function]}
            onScrollBeginDrag={[Function]}
            onScrollEndDrag={[Function]}
            onViewableItemsChanged={[Function]}
            removeClippedSubviews={false}
            renderItem={[Function]}
            scrollEventThrottle={50}
            showsHorizontalScrollIndicator={false}
            stickyHeaderIndices={Array []}
            updateCellsBatchingPeriod={50}
            viewabilityConfig={
              Object {
                "itemVisiblePercentThreshold": 75,
                "waitForInteraction": true,
              }
            }
            viewabilityConfigCallbackPairs={
              Array [
                Object {
                  "onViewableItemsChanged": [Function],
                  "viewabilityConfig": Object {
                    "itemVisiblePercentThreshold": 75,
                    "waitForInteraction": true,
                  },
                },
              ]
            }
            windowSize={21}
          >
            <View />
          </RCTScrollView>
        </View>
        <View
          style={
            Object {
              "flexDirection": "row",
              "height": 30,
              "justifyContent": "center",
              "width": "100%",
            }
          }
        />
      </View>
    </View>
  </RCTScrollView>
</RCTSafeAreaView>
`;
